src/proto/mgmt/event.proto

```diff
@@ -0,0 +1,72 @@
+//
+// (C) Copyright 2020 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+// This file defines the messages used by gRPC to communicate events.
+// TODO: find a way to sensibly share definitions in duplicate file srv/event.
+
+syntax = "proto3";
+package mgmt;
+
+// RankStateEventInfo defines extended fields for rank state change related events.
+message RankStateEventInfo {
+	uint32 instance = 1;	// Control-plane harness instance index.
+	bool errored = 2;	// Rank in error state.
+	string error = 3;	// Message associated with error.
+}
+
+// PoolSvcEventInfo defines extended fields for pool service change events.
+message PoolSvcEventInfo {
+	string pool_uuid = 1;		// Pool UUID.
+	repeated uint32 svc_reps = 2;	// Pool service replica ranks.
+	int32 raft_leader_term = 3;	// Raft leadership term.
```

Comment:
It would be better to replace

    int32 raft_leader_term = 3;	// Raft leadership term.

with

    uint64 version = 3; // Info version.

The pool service code don't have a way to fill this field yet; I think we'll implement it with the Raft log index of the corresponding membership change entry (or a Raft log index that >= such, provided that there's no other membership change entry in between). Once we have it implemented, the MS can start ignoring incoming infos with lower versions than the version it already has. [May be regarded of out of scope of this PR.]

Additionally, the pool service will send the latest info to the MS if it is not 100% sure that the MS already has the info. E.g., upon NOTIFY errors (using a separate ULT), pool service crashes (whenever a new leader steps up, or track persistently the version of the last successful NOTIFY).  [May be regarded of out of scope of this PR.]

Reply:
As to implementing "version" using "Raft log index", on second thought, maybe it's okay to use "Raft leadership term", as within one leadership term, out-of-order events feel unlikely. That said, I still think it'd be better to just use a uint64 version field; here we don't need the concept of "Raft" and "term".

Reply:
thanks for the feedback, I think population of the field and info notify hints work can be deferred to subsequent PRs but I will change the placeholder field name and type to version and uint64 respectively in this PR

Reply:
Thanks for considering, Tom.